code/script.js

```diff
@@ -222,104 +243,151 @@ const generateBoard = () => {
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  } else if (category === 'other') {
+    if (keep) {
+      alert(`Correct! Keep characters with ${value}`)
+    } else {
+      alert(`Wrong!! Remove characters with ${value}`)
+    }  
+}
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+// compare with secret person
+if (category === 'hair' || category === 'eyes') {
+  if (keep) {
+  charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+} else {
+  charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+}
+}
 
-  // Invoke a function to redraw the board with the remaining people.
+if (category === 'accessories' || category === 'other') {
+  if (keep) {
+  charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+} else {
+  charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+}
+}
+generateBoard() // set new board with the remaining characters. 
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
+// If the player wants to guess, invoke the checkMyGuess function.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  let playerGuess = confirm("Do you want to make a guess on this person?")
+
+  if (playerGuess === true) {
+    checkMyGuess(personToConfirm)
+  } else {
+    alert ("Make a new guess...")
+  }
+  }
 
-// If you confirm, this function is invoked
+// invoked when user confirm final guess // Show the final win or lose section
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+
+  if (personToCheck === secret.name) {
+    soundWon();
+    winOrLoseText.innerText = `Yay! ${secret.name} is the correct! Want to play again?` 
+    winOrLose.style.display = 'flex' // hiding the board
+
+  } else {
+    soundLost();
+    winOrLoseText.innerText = `Oh... It is not ${personToCheck}. Right answer is ${secret.name}. Play again?`
+    winOrLose.style.display = 'flex' // hiding the board
+  }
 }
+start()  //HTML
+
+// Event listeners
+restartButton.addEventListener('click', start)  // restart current game
+questions.addEventListener('change', selectQuestion)  // dropdown with selections 
+findOutButton.addEventListener('click', checkQuestion && updateGuesses) 
+playAgainButton.addEventListener('click', start) // back to homepage)
+playButton.addEventListener('click',closeOverlay);
```

Comment:
I think here where better to put a 'submit' listener, because if you press the key enter in the name input bar, the browser refresh. And put the input bar id better than the button.